src_sycl/renderer_gl.cpp

```diff
@@ -297,4 +310,60 @@ void RendererGL::render(glm::mat4 proj_mat, glm::mat4 view_mat) {
    glDrawArrays(GL_TRIANGLES, 0, 3);
 }
 
+std::vector<float> RendererGL::gaussKernel(const float sigma,
+                                           const int halfwidth) {
+   float sigma_factor = 1.0 / (sigma * sqrt(2 * glm::pi<float>()));
+
+   auto sigma_fun = [sigma, sigma_factor, n = 0]() mutable {
+      float sigma_val =
+          sigma_factor * std::exp(-std::pow(static_cast<float>(n), 2) /
+                                  (2 * std::pow(sigma, 2)));
+      n++;
+      return sigma_val;
+   };
+
+   std::vector<float> result(halfwidth);
+   std::generate(result.begin(), result.end(), sigma_fun);
+
+   // Normalize the Gaussian kernel
+   float halfnorm = std::accumulate(result.begin() + 1, result.end(), 0.0);
+   float norm = 2 * halfnorm + result[0];
+
+   std::transform(result.begin(), result.end(), result.begin(),
+                  [&norm](auto &val) { return val / norm; });
```

Comment:
```suggestion
                  [norm](auto val) { return val / norm; });
```

not sure why using references here

Reply:
good point thanks